format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingInputFormatProvider.java

```diff
@@ -63,6 +65,28 @@ protected void validate() {
     // no-op
   }
 
+  protected Schema addPathField(Schema dataSchema, FailureCollector failureCollector) {
+    return addPathField(failureCollector, dataSchema, conf.getPathField());
+  }
+
+  public static Schema addPathField(FailureCollector failureCollector, Schema dataSchema, @Nullable String pathField) {
+    if (pathField == null) {
+      return dataSchema;
+    }
+    List<Schema.Field> fields = new ArrayList<>(dataSchema.getFields());
+    for (Schema.Field field : fields) {
+      if (pathField.equals(field.getName())) {
+        failureCollector.addFailure(
+          String.format("Path Field %s already exists in the data schema ", pathField),
+            "Please provide a field that does not already exist")
```

Comment:
fix indentation here. This should be one of the following:

```
    failureCollector.addFailure(String.format("Path Field %s already exists in the data schema ", pathField), 
                                "Please provide a field that does not already exist")
      .withConfigProperty("pathField");
```
or 

```
    failureCollector.addFailure(
      String.format("Path Field %s already exists in the data schema ", pathField), 
      "Please provide a field that does not already exist")
      .withConfigProperty("pathField");
```

Reply:
Done